examples/2d/system_fonts.rs

```diff
@@ -0,0 +1,73 @@
+//! Demonstrates searching for a system font at runtime.
+//!
+//! This example cycles through all available fonts, showing their name in said
+//! font.
+//! Pressing the spacebar key will select the next font.
+
+use bevy::{platform_support::collections::HashSet, prelude::*, text::FontLibrary};
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .add_systems(Startup, setup)
+        .add_systems(Update, update)
+        .run();
+}
+
+/// Marker for the text we will update.
+#[derive(Component)]
+struct FontName;
+
+fn setup(mut commands: Commands, asset_server: Res<AssetServer>) {
+    commands.spawn(Camera2d);
+    commands.spawn((
+        Text2d::new("Press Space to change the font"),
+        TextFont {
+            // We load a fall-back font, since we don't know that every platform
+            // even has system fonts to use.
+            font: asset_server.load("fonts/FiraSans-Bold.ttf"),
+            font_size: 50.0,
+            ..default()
+        },
+        FontName,
+    ));
+}
+
+fn update(
+    // FontLibrary provides access to all fonts loaded by the cosmic_text backend.
+    // With the `system_font` feature enabled, this includes fonts installed on
+    // the end-user's device.
+    mut fonts: FontLibrary,
+    mut query: Single<(&mut Text2d, &mut TextFont), With<FontName>>,
+    keyboard_input: Res<ButtonInput<KeyCode>>,
+    // Want to make sure we don't show the same font twice
+    mut used: Local<HashSet<Box<str>>>,
+) {
+    if keyboard_input.just_pressed(KeyCode::Space) {
+        let mut name = default();
+
+        // The primary way to find a font is through FontLibrary::find,
+        // which iterates over all loaded fonts, and returns the first that
+        // satisfies the provided predicate.
+        // In this example, we're just looking for any font we haven't already
+        // shown.
+
+        let Some(font) = fonts.find(|font| {
+            let family_name = font.families[0].0.as_str();
+
+            if used.contains(family_name) {
+                return false;
+            }
+
+            used.insert(Box::from(family_name));
+            name = String::from(family_name);
+            true
+        }) else {
+            *query.0 = Text2d::new("No more fonts to show!");
+            return;
+        };
+
+        *query.0 = Text2d::new(name);
+        query.1.font = font;
+    }
```

Comment:
Nitpick - I think there's an idiomatic way to avoid some returns here.
```suggestion
    if keyboard_input.just_pressed(KeyCode::Space) {
        let mut name = default();
        
        // The primary way to find a font is through FontLibrary::find,
        // which iterates over all loaded fonts, and returns the first that
        // satisfies the provided predicate.
        // In this example, we're just looking for any font we haven't already
        // shown.
        
        let next_font = fonts.find(|font|{
            let family_name = font.families[0].0.as_str();
            if used.contains(family_name) {
                false
            } else {
                used.insert(Box::from(family_name));
                name = String::from(family_name);
                true
            }
        });
        
        match next_font {
            Some(font) => {
                *query.0 = Text2d::new(name);
                query.1.font = font;
            },
            None => {
                *query.0 = Text2d::new("No more fonts to show!");
            },
        }
    }
```

Reply:
Actually I think for the end, `clippy` usually prefers `if let ... else` also but I did this in github comments not my editor XD

Reply:
An example alternate approach that I think ends up a lot more readable, if as in my other comment more info was returned from `.find`
```rs
    if keyboard_input.just_pressed(KeyCode::Space) {
        // The primary way to find a font is through FontLibrary::find,
        // which iterates over all loaded fonts, and returns the first that
        // satisfies the provided predicate.
        // In this example, we're just looking for any font we haven't already
        // shown.
        
        let next_font = fonts.find(|font|{
            let family_name = font.families[0].0.as_str();
            used.contains(family_name)
        });
        
        if let Some((handle, face_info) = next_font {
            let family_name = face_info.families[0].0.as_str();
            *query.0 = Text2d::new(family_name);
            query.1.font = handle;
        } else {
            *query.0 = Text2d::new("No more fonts to show!");
        }
    }
```